src/rnnlm/rnnlm-utils.cc

```diff
@@ -0,0 +1,233 @@
+#include "rnnlm/rnnlm-utils.h"
+
+namespace kaldi {
+namespace rnnlm {
+
+vector<string> SplitByWhiteSpace(const string &line) {
+  std::stringstream ss(line);
+  vector<string> ans;
+  string word;
+  while (ss >> word) {
+    ans.push_back(word);
+  }
+  return ans;
+}
+
+unordered_map<string, int> ReadWordlist(string filename) {
+  unordered_map<string, int> ans;
+  ifstream ifile(filename.c_str());
+  string word;
+  int id;
+
+  while (ifile >> word >> id) {
+    ans[word] = id;
+  }
+  return ans;
+}
+
+NnetExample GetEgsFromSent(const vector<int>& word_ids_in, int input_dim,
+                           const vector<int>& word_ids_out, int output_dim) {
+  SparseMatrix<BaseFloat> input_frames(word_ids_in.size(), input_dim);
+
+  for (int j = 0; j < word_ids_in.size(); j++) {
+    vector<std::pair<MatrixIndexT, BaseFloat> > pairs;
+    pairs.push_back(std::make_pair(word_ids_in[j], 1.0));
+    SparseVector<BaseFloat> v(input_dim, pairs);
+    input_frames.SetRow(j, v);
+  }
+
+  NnetExample eg;
+  eg.io.push_back(nnet3::NnetIo("input", 0, input_frames));
+
+  Posterior posterior;
+  for (int i = 0; i < word_ids_out.size(); i++) {
+    vector<std::pair<int32, BaseFloat> > p;
+    p.push_back(std::make_pair(word_ids_out[i], 1.0));
+    posterior.push_back(p);
+  }
+
+  eg.io.push_back(nnet3::NnetIo("output", output_dim, 0, posterior));
+  return eg;
+}
+
+void SelectWithoutReplacement(const vector<BaseFloat> &u, int n, vector<int> *out) {
```

Comment:
as I said before, this should have 'sample' in the name, not 'select' (sample implies randomness).